uwblueprint/abtc
# Shifts-related model changes, routes, services and frontend UI

## Notion ticket link

[Create the reusable card for the “Your Shifts” on the volunteer dashboard](https://www.notion.so/uwblueprintexecs/Create-the-reusable-card-for-the-Your-Shifts-on-the-volunteer-dashboard-b9b52fde416f48bcb959c62914528a22?pvs=4)



## Implementation description
* Refactored the way shifts and users are linked to each other (to bypass oddities with how prisma-mongodb combination deals with many-to-many relations)
* Added GET and POST APIs for querying shifts that an admin has requested & linking that
* Added GET and POST APIs for querying shifts that a volunteer is assigned to & updating the link between them
* Created frontend UI and API calls o display shift info for a user in cards



## Steps to test
1. Log into the app using your credentials
2. Find your authId in the development collection on mongodb
3. Find one or more random ObjectIds for serviceRequests on mongodb, and assign yourself these shifts in postman: (replace with your authId & the shift's ObjectId)
-> ideally choose a serviceRequest that has a shiftEndTime (the older ones didn't have this)
<img width="850" alt="image" src="https://github.com/uwblueprint/abtc/assets/67832534/fa2c6a78-6bfa-41e9-9120-a9b62b85a155">
5. You can check that it worked using the GET (replace with your userId)
<img width="849" alt="image" src="https://github.com/uwblueprint/abtc/assets/67832534/d8a35467-abff-49b0-837f-626c0c29afe0">
6. Go to localhost:3000/shifts to see the frontend!


## What should reviewers focus on?
* 


## Checklist

## Remaining to-dos:
- We should protect at least the POST route so only admins can assign
- Add a delete API for removing assignees from a shift
